gadcp/madcp.py

```diff
@@ -846,6 +845,237 @@ def _regrid_amp(self, ens, method="linear"):
+        for i in tqdm(range(nens)):
+            ens = self.m.read(start=ens_idxs[i], stop=ens_idxs[i + 1])
+            idx0 = write_idxs[i]
+            idx1 = write_idxs[i + 1]
+
+            if ens is not None:
+                # I pulled this out of read_ensembles because we need to calculate depth for _ave2nc to work.
+                ens.dday_orig = ens.dday
+                ens.dday = self._correct_dday(ens.dday_orig)
+                if self._pressure_provided is not None:
+                    # Replace pressure if provided
+                    ens.pressure = self._external_pressure_to_dat(ens)
+                else:
+                    ens.pressure = self._scale_pycurrents_pressure(ens)
+                sign = -1 if self.orientation == "up" else 1
+                pdepth = seawater.depth2(ens.pressure, self.lat)
+                ens.depth = pdepth[:, np.newaxis] + sign * ens.dep
+
+                if binmap:
+                    self._binmap_all_beams(ens)
+                    # Now we have to recalculate xyze with the binmapped data.
+                    self._calculate_xyze(ens)
+
+                self._edit(ens)  # modifies xyze
+                self._to_enu(ens)  # transform to earth coords (east, north, up)
+
+            else:
+                uvwe[idx0:idx1] = np.ma.masked
+                amp[idx0:idx1] = np.ma.masked
+                # pressure[idx0:idx1] = np.ma.masked
+                # temperature[idx0:idx1] = np.ma.masked
+                continue
+
+            uvwe[idx0:idx1] = ens.enu
+
+            # pgi = 100 * ens.enu_grid[..., 0].count(axis=0) // nprofs
+            # pg[i] = pgi.astype(np.int8)
+            amp[idx0:idx1] = ens.amp.mean(axis=-1)  # Average over beams... why?
+
+        self.ave = Bunch(
+            u=uvwe[..., 0],
+            v=uvwe[..., 1],
+            w=uvwe[..., 2],
+            e=uvwe[..., 3],
+            pg=pg,
+            amp=amp,
+            temperature=temperature,
+            pressure=pressure,
+            # npings=npings,
+            dday=dday,
+            yearbase=self.yearbase,
+            dep=ens.dep,  # <<<<----- BAD!!! This a fudge because I don't want to calculated a depth vector. We use the depth vector from the last ensemble.
+            editparams=self.editparams,
+            tgridparams=self.tgridparams,
+            # dgridparams=self.dgridparams,
+            magdec=self.magdec,
+            lon=self.lon,
+            lat=self.lon,
+        )
+
+        self._ave2nc()
+
+        # Add some more info.
+        self.ds.attrs["orientation"] = self.orientation
+        self.ds.attrs["magdec"] = self.magdec
+        for att in ["max_e", "max_e_deviation", "min_correlation"]:
+            self.ds.attrs[att] = self.editparams[att]
+
+        # Add meta data if provided.
+        if self.meta_data is not None:
+            for k, v in self.meta_data.items():
+                self.ds.attrs[k] = v
+        self.ds.attrs["proc time"] = np.datetime64("now").astype("str")
```

Comment:
You could replace the following lines with the little function that now adds meta data and CF compliant attributes:
```suggestion
    self._add_meta_data_to_ds()
```

Reply:
I guess this speaks to the larger issue that we should merge some of the beam editing / averaging functions. We can discuss further in #15 or maybe chat in person about it.